conans/client/build/cmake.py

```diff
@@ -184,6 +184,11 @@ def configure(self, args=None, defs=None, source_dir=None, build_dir=None,
                 self._conanfile.run(command)
 
     def build(self, args=None, build_dir=None, target=None):
+        if not self._conanfile.should_build:
+            return
+        self._build(args, build_dir, target)
+
+    def _build(self, args=None, build_dir=None, target=None):
         if not self._conanfile.should_build:
```

Comment:
Remove this in `_build`?

Reply:
Exactly that is the point of the PR, this check shouldn't be here.

Reply:
Please add a test that checks this PR is actually fixing this issue.